clients/client-python/gravitino/filesystem/gvfs.py

```diff
@@ -465,134 +489,105 @@ def get_file(self, rpath, lpath, callback=None, outfile=None, **kwargs):
-        read_lock = self._cache_lock.gen_rlock()
-        try:
-            read_lock.acquire()
-            cache_value: Tuple[Fileset, AbstractFileSystem, StorageType] = (
-                self._cache.get(identifier)
-            )
-            if cache_value is not None:
-                actual_path = self._get_actual_path_by_ident(
-                    identifier,
-                    cache_value[0],
-                    cache_value[1],
-                    cache_value[2],
-                    virtual_path,
-                )
-                return FilesetContext(
-                    identifier,
-                    cache_value[0],
-                    cache_value[1],
-                    cache_value[2],
-                    actual_path,
-                )
-        finally:
-            read_lock.release()
-
-        write_lock = self._cache_lock.gen_wlock()
-        try:
-            write_lock.acquire()
-            cache_value: Tuple[Fileset, AbstractFileSystem] = self._cache.get(
-                identifier
-            )
-            if cache_value is not None:
-                actual_path = self._get_actual_path_by_ident(
-                    identifier,
-                    cache_value[0],
-                    cache_value[1],
-                    cache_value[2],
-                    virtual_path,
-                )
-                return FilesetContext(
-                    identifier,
-                    cache_value[0],
-                    cache_value[1],
-                    cache_value[2],
-                    actual_path,
-                )
-            fileset: Fileset = self._load_fileset_from_server(identifier)
-            storage_location = fileset.storage_location()
-            if storage_location.startswith(f"{StorageType.HDFS.value}://"):
-                fs = ArrowFSWrapper(HadoopFileSystem.from_uri(storage_location))
-                storage_type = StorageType.HDFS
-            elif storage_location.startswith(f"{StorageType.LOCAL.value}:/"):
-                fs = LocalFileSystem()
-                storage_type = StorageType.LOCAL
-            else:
-                raise GravitinoRuntimeException(
-                    f"Storage under the fileset: `{identifier}` doesn't support now."
-                )
-            actual_path = self._get_actual_path_by_ident(
-                identifier, fileset, fs, storage_type, virtual_path
-            )
-            self._cache[identifier] = (fileset, fs, storage_type)
-            context = FilesetContext(identifier, fileset, fs, storage_type, actual_path)
-            return context
-        finally:
-            write_lock.release()
+        catalog_ident: NameIdentifier = NameIdentifier.of(
+            self._metalake, identifier.namespace().level(1)
+        )
+        fileset_catalog = self._get_fileset_catalog(catalog_ident)
+        assert (
+            fileset_catalog is not None
+        ), f"Loaded fileset catalog: {catalog_ident} is null."
```

Comment:
Do we need an exception to handle this, or just `assert` is enough?

Reply:
Change to throw an exception here, I think it's better.